### PR TITLE
Type annotations for archinfo

### PR DESCRIPTION
**Pull request by fmagin**
Discussion in angr/pyvex#198

### CODE REVIEW

File: archinfo/arch.py [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import logging
+from typing import Dict, List, Tuple
+
 import struct as _struct
 import platform as _platform
 import re
 from archinfo.archerror import ArchError
+from pyvex import RegisterOffset, RegisterName
```
Comment (twizmwazin) [inline archinfo/arch.py]: We shouldn't be using pyvex in archinfo.

File: setup.py [hunk 2]
```diff
@@ -15,4 +15,7 @@
     version='8.20.1.7',
     python_requires='>=3.5',
     packages=packages,
+    package_data={'archinfo': ['py.typed']},
+    test_suite='nose.collector',
+    tests_require=['nose'],
```
Comment (twizmwazin) [inline setup.py]: I'm not sure that we want these two lines, at least as part of the PR.

Comment (fmagin) [reply setup.py]: That was probably some rebase related messup and definitely not intentional, will fix.

File: archinfo/arch.py (hunk 1 above)
Comment (fmagin) [reply archinfo/arch.py]: This raises the issue of where to put those type aliases. They need to be unique, so it is not possible for each module to define it themselves. The only other option that comes to mind right now would be a dedicated module for type aliases like this. Declaring that some integer is a RegisterOffset and not a TempVar is one of the most useful benefits of type annotations for this part of the codebase, so I definitely want this somewhere.
The meaning mapping registername <-> register offset comes from Vex it seems reasonable to me to get it from pyvex. What issues would be introduced by this?

Comment (rhelmot) [reply archinfo/arch.py]: archinfo just needs to still work when pyvex isn't installed, just without the pyvex features. you can put this in an optional idiom and make them None if pyvex throws an ImportError.

Alternately, the type aliases can just live in archinfo.

**Pull request merged**